src/metkit/api/metkit_c.cc

```diff
@@ -0,0 +1,348 @@
+        eckit::Log::error() << "METKIT Error: " << e.what() << std::endl;
+        g_current_error_string = e.what();
+        return METKIT_ERROR;
+    }
+    catch (const std::exception& e) {
+        eckit::Log::error() << "Unknown Error: " << e.what() << std::endl;
+        g_current_error_string = e.what();
+        return METKIT_ERROR_UNKNOWN;
+    }
+    catch (...) {
+        eckit::Log::error() << "Unknown Error!" << std::endl;
+        g_current_error_string = "<unknown>";
+        return METKIT_ERROR_UNKNOWN;
+    }
+}
+
+extern "C" {
+
+// -----------------------------------------------------------------------------
+//                           HELPERS
+// -----------------------------------------------------------------------------
+
+int metkit_version(const char** version) {
+    *version = metkit_version_str();
+    return METKIT_SUCCESS;
+}
+
+int metkit_vcs_version(const char** sha1) {
+    *sha1 = metkit_git_sha1();
+    return METKIT_SUCCESS;
+}
+
+int metkit_initialise() {
+    return tryCatch([] {
+        static bool initialised = false;
+
+        if (initialised) {
+            eckit::Log::warning()
+                << "Initialising Metkit library twice" << std::endl;
+        }
+
+        if (!initialised) {
+            const char* argv[2] = {"metkit-api", 0};
+            eckit::Main::initialise(1, const_cast<char**>(argv));
+            initialised = true;
+        }
+    });
+}
+
+// -----------------------------------------------------------------------------
+//                           PARSING
+// -----------------------------------------------------------------------------
+
+int metkit_parse_marsrequest(const char* str, metkit_requestiterator_t** requests, bool strict) {
+    return tryCatch([requests, str, strict] {
+        ASSERT(requests);
+        ASSERT(str);
+        std::istringstream in(str);
+        *requests = new metkit_requestiterator_t(metkit::mars::MarsRequest::parse(in, strict));
+    });
+}
+
+// -----------------------------------------------------------------------------
+//                           REQUEST
+// -----------------------------------------------------------------------------
+
+int metkit_new_marsrequest(metkit_marsrequest_t** request) {
+    return tryCatch([request] {
+        ASSERT(request);
+        *request = new metkit_marsrequest_t();
+    });
+}
+
+int metkit_free_marsrequest(const metkit_marsrequest_t* request) {
+    return tryCatch([request] {
+        ASSERT(request);
+        delete request;
+    });
+}   
+
+int metkit_marsrequest_add(metkit_marsrequest_t* request, const char* param, const char* values[], int numValues) {
+    return tryCatch([request, param, values, numValues] {
+        ASSERT(request);
+        ASSERT(param);
+        ASSERT(values);
+        std::string n(param);
+        std::vector<std::string> vv;
+        for (int i = 0; i < numValues; i++) {
+            vv.push_back(std::string(values[i]));
+        }
```

Comment:
Its a good idea to use reserve as you already know the number of strings.
Also you can skip the loop.
```suggestion
        std::vector<std::string> vv;
        vv.reserve(numValues);
        std::copy(values, values + numValues, std::back_inserter(vv));
```